File: gui-cpp/src/pages/identitypage.cpp
/**
 * 页2: 身份配置 — 实现。
 */
#include "identitypage.h"
#include <QVBoxLayout>
#include <QGridLayout>
#include <QGroupBox>
#include <QRadioButton>

IdentityPage::IdentityPage(QWidget *parent) : QWidget(parent) {
    auto *layout = new QVBoxLayout(this);
    layout->setSpacing(12);

    // 原始身份
    auto *origGroup = new QGroupBox("原始身份");
    auto *origGrid = new QGridLayout(origGroup);
    const char *keys[] = {"Vendor ID", "Product Code", "Revision", "Device Name"};
    for (int i = 0; i < 4; ++i) {
        origGrid->addWidget(new QLabel(QString(keys[i]) + ":"), i, 0);
        m_origLabels[i] = new QLabel("-");
        m_origLabels[i]->setStyleSheet("color: #888;");
        origGrid->addWidget(m_origLabels[i], i, 1);
    }
    layout->addWidget(origGroup);

    // 新身份
    auto *newGroup = new QGroupBox("新身份 (覆盖)");
    auto *newGrid = new QGridLayout(newGroup);
    auto addRow = [&](int row, const char *label) -> QLineEdit* {
        newGrid->addWidget(new QLabel(label), row, 0);
        auto *edit = new QLineEdit;
        newGrid->addWidget(edit, row, 1);
        return edit;
    };
    m_vendorEdit   = addRow(0, "Vendor ID:");
    m_productEdit  = addRow(1, "Product Code (hex):");
    m_revisionEdit = addRow(2, "Revision (hex):");
    m_nameEdit     = addRow(3, "Device Name:");
    layout->addWidget(newGroup);

    // 模式选择
    auto *modeGroup = new QGroupBox("模式");
    auto *modeLayout = new QHBoxLayout(modeGroup);
    m_modeGroup = new QButtonGroup(this);
    const char *modes[] = {"学习模式", "实验室克隆", "产品模式"};
    for (int i = 0; i < 3; ++i) {
        auto *rb = new QRadioButton(modes[i]);
        m_modeGroup->addButton(rb, i);
        modeLayout->addWidget(rb);
    }
    m_modeGroup->button(0)->setChecked(true);
    layout->addWidget(modeGroup);
    layout->addStretch();
}

void IdentityPage::setOriginalIdentity(const QVariantMap &id) {
    m_origLabels[0]->setText(id["Vendor ID"].toString());
    m_origLabels[1]->setText(id["Product Code"].toString());
    m_origLabels[2]->setText(id["Revision"].toString());
    m_origLabels[3]->setText(id["Device Name"].toString());
}

QVariantMap IdentityPage::getConfig() const {
    QVariantMap m;
    m["vendor_id"]    = m_vendorEdit->text().trimmed();
    m["product_code"] = m_productEdit->text().trimmed();
    m["revision"]     = m_revisionEdit->text().trimmed();
    m["device_name"]  = m_nameEdit->text().trimmed();
    const char *modeNames[] = {"learning", "lab", "product"};
    m["mode"] = modeNames[m_modeGroup->checkedId()];
    return m;
}

File: gui/widgets/identity_page.py
"""
Page 2: Identity Configuration (dev doc §12.2)

Shows original device identity, lets user override for output.
Mode selection: Learning / Lab Clone / Product.
"""
from PyQt5.QtWidgets import (
    QWidget, QVBoxLayout, QGridLayout,
    QLabel, QLineEdit, QGroupBox, QRadioButton,
    QButtonGroup, QHBoxLayout,
)


class IdentityPage(QWidget):
    """Device identity configuration page."""

    def __init__(self, parent=None):
        super().__init__(parent)
        self._init_ui()

    def _init_ui(self):
        layout = QVBoxLayout(self)
        layout.setSpacing(12)

        # ── Original identity (read-only) ──────
        orig_group = QGroupBox("Original Identity")
        orig_grid = QGridLayout(orig_group)
        self._orig_labels = {}
        for i, key in enumerate(["Vendor ID", "Product Code", "Revision", "Device Name"]):
            orig_grid.addWidget(QLabel(f"{key}:"), i, 0)
            lbl = QLabel("-")
            lbl.setStyleSheet("color: gray;")
            orig_grid.addWidget(lbl, i, 1)
            self._orig_labels[key] = lbl
        layout.addWidget(orig_group)

        # ── New identity (editable) ────────────
        new_group = QGroupBox("New Identity (override)")
        new_grid = QGridLayout(new_group)
        self._vendor_edit = self._add_row(new_grid, 0, "Vendor ID:")
        self._product_edit = self._add_row(new_grid, 1, "Product Code (hex):")
        self._revision_edit = self._add_row(new_grid, 2, "Revision (hex):")
        self._name_edit = self._add_row(new_grid, 3, "Device Name:")
        layout.addWidget(new_group)

        # ── Mode selection (§12.2) ─────────────
        mode_group = QGroupBox("Mode")
        mode_layout = QHBoxLayout(mode_group)
        self._mode_group = QButtonGroup(self)
        for i, (text, tooltip) in enumerate([
            ("Learning", "Default — safe for development"),
            ("Lab Clone", "Clone existing device identity"),
            ("Product", "Production — use real identity"),
        ]):
            rb = QRadioButton(text)
            rb.setToolTip(tooltip)
            self._mode_group.addButton(rb, i)
            mode_layout.addWidget(rb)
        self._mode_group.button(0).setChecked(True)  # default: Learning
        layout.addWidget(mode_group)

        layout.addStretch()

    def _add_row(self, grid, row, label):
        lbl = QLabel(label)
        edit = QLineEdit()
        grid.addWidget(lbl, row, 0)
        grid.addWidget(edit, row, 1)
        return edit

    # ── Public API ─────────────────────────────

    def set_original_identity(self, identity: dict):
        """Populate the original identity display."""
        for key, lbl in self._orig_labels.items():
            lbl.setText(str(identity.get(key, "-")))

    def get_config(self) -> dict:
        """Return user's identity overrides."""
        mode_map = {0: "learning", 1: "lab", 2: "product"}
        mode_id = self._mode_group.checkedId()
        return {
            "vendor_id": self._vendor_edit.text().strip(),
            "product_code": self._product_edit.text().strip(),
            "revision": self._revision_edit.text().strip(),
            "device_name": self._name_edit.text().strip(),
            "mode": mode_map.get(mode_id, "learning"),
        }
